### PR TITLE
refactor: apply webpack defaults

### LINKED ISSUE #7: webpack-defaults upgrade
Addition of `webpack-defaults` & associated refactoring as a part of the next `Major` release

Issue exists for status tracking across the organization. 

>Please do not close

### PR DESCRIPTION
**Pull request by joshwiens**
Intended to be merged & released as a part of `1.0.0` on a beta `dist-tag` once this has been finished and properly tested.

BREAKING CHANGE: 

Enforces NodeJS > 4.3 via engines 

Closed #7

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,5 +36,37 @@
...
+    "pre-commit": "^1.2.2",
+    "standard-version": "^4.0.0",
+    "webpack-defaults": "^0.4.5"
+  },
+  "engines": {
+    "node": ">= 4.3 < 5.0.0 || >= 5.10"
+  },
+  "files": [
+    "dist"
+  ],
```
Comment (michael-ciniawsky) [inline package.json]:
```
"repository": {
    "type": "git",
    "url": "git+https://github.com/webpack-contrib/null-loader.git"
  },
  "bugs": {
    "url": "https://github.com/webpack-contrib/null-loader/issues"
  },
  "homepage": "https://webpack.js.org",
```

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default function () {
+  this.cacheable();
```
Comment (michael-ciniawsky) [inline src/index.js]:
```diff
- this.cacheable();
```

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  return '// empty (null-loader)';
+}
+
+export function pitch() {
+  this.cacheable();
```
Comment (michael-ciniawsky) [inline src/index.js]:
```diff
- this.cacheable();
```

**Review by michael-ciniawsky — CHANGES_REQUESTED** [review]

**Review by michael-ciniawsky — APPROVED** [review]

**Comment by joshwiens** [comment]
So if I find a good way to test `nothing` do I win some sort of prize?

**Comment by michael-ciniawsky** [comment]
The **all and nothing** award is yours 🥇 😛

**Review by shellscape — APPROVED** [review]

**Pull request merged**